battleGame.js

```diff
@@ -1,15 +1,59 @@
+
+
 // 1. Create attack function below.  This will take the following parameters:
 // attackingPlayer, defendingPlayer, baseDamage, variableDamage
 
+// let variableDamage = Math.random()*10
+// const baseDamage = 1;
+// const attack = function(attackingPlayer, defendingPlayer, baseDamage, variableDamage){
+  
+//   // this will allow playability vs one player dominating 
+//   if(variableDamage>5){
+//     variableDamage=5;
+//   }
+//   defendingPlayer - (baseDamage + variableDamage)
+//   let totalDamage = baseDamage + variableDamage
 
+//   console.log(`
+//   Attacking Player: ${attackingPlayer.name}
+//   Defending Player: ${defendingPlayer.name}
+  
+//   Defending Player recived ${totalDamage} in damage
+//   Defending Player Health: ${defendingPlayer.health}
+  
+//   Attacking Player Health: ${attackingPlayer.health}`)
+// }
 
 // 2. Create player1 and player2 objects below
 // Each should have a name property of your choosing, and health property equal to 10
 
-
+let player1 = {
+    name: 'Player 1',
+    health: 10,
+}
+let player2 = {
+  name: 'Player 2',
+  health: 10,
+}
 
 // 3. Refactor attack function to an arrow function.  Comment out function above.
+const attack = (attackingPlayer, defendingPlayer, baseDamage, variableDamage) =>{
+  // this line avoids single move wins
+  if(variableDamage > 9){
+    variableDamage = Math.floor(Math.random() *10)
+  }
+  defendingPlayer.health -= (baseDamage + variableDamage)
+  let totalDamage = baseDamage + variableDamage
 
+  console.log(`
```

Comment:
Just a key idea I wanna point out . . . you want to _return_ a string, not `console.log()` . ..  because that's what line 69 is doing! But, as you can see when you run the program, line 69 logs `undefined` because this attack() function does not return any value. 

Keep in mind that console.log() is a tool for developers to see their code while they write. But it's far more important to actually be returning the values our functions calculate.